### PR TITLE
migrate more test utils used by smoke tests to use rest api client

### PR DESCRIPTION
**Pull request by xli**
## Motivation

#9193 
Migrate smoke tests to use rest api client:
1. Added getting epoch configuration to rest client
2. Update smoke test to use rest api client to get epoch
3. Migrate forge interface public usage context and chain info to use rest client
4. Migrate nft smoke test to use rest api client
5. [api] add accumulator_root_hash to onchain Transaction for forge fork_check

## Test Plan

CI build

## Related PRs

#10032

### CODE REVIEW

File: testsuite/forge-cli/src/main.rs [hunk 1]
```diff
@@ -444,14 +444,22 @@ impl Test for FundAccount {
...
 impl PublicUsageTest for FundAccount {
     fn run<'t>(&self, ctx: &mut PublicUsageContext<'t>) -> Result<()> {
-        let client = ctx.client();
+        let runtime = Runtime::new().unwrap();
+        runtime.block_on(self.async_run(ctx))
+    }
+}
+
+impl FundAccount {
+    async fn async_run(&self, ctx: &mut PublicUsageContext<'_>) -> Result<()> {
```
Comment (bmwill) [inline testsuite/forge-cli/src/main.rs:453]: Yeah i'll need to investigate just having an async run function.

Comment (zihaoccc) [reply testsuite/forge-cli/src/main.rs:453]: I think we can plan to replace runtime block in forge to use async fn

File: testsuite/forge/src/runner.rs [hunk 2]
```diff
@@ -282,17 +283,20 @@ impl<'cfg, F: Factory> Forge<'cfg, F> {
 
             // Run NFTPublicUsageTests
             if !self.tests.nft_public_usage_tests.is_empty() {
```
Comment (bmwill) [inline testsuite/forge/src/runner.rs:285]: Interesting, i must have missed when this was added

**Review by bmwill — APPROVED** [review]

**Review by bmwill — APPROVED** [review]

**Review by bmwill — DISMISSED** [review]

**Pull request merged**